src/Plugin/facets/widget/MultiselectWidget.php

```diff
@@ -16,7 +24,127 @@
+      $plugin_definition,
+      $container->get('entity_type.manager'),
+      $container->get('entity_field.manager')
+    );
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  public function __construct(array $configuration, $plugin_id, $plugin_definition, EntityTypeManagerInterface $entity_type_manager, EntityFieldManagerInterface $entity_field_manager) {
+    parent::__construct($configuration, $plugin_id, $plugin_definition);
+    $this->entityTypeManager = $entity_type_manager;
+    $this->entityFieldManager = $entity_field_manager;
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  public function getDefaultValuesLabel(FacetInterface $facet, ListSourceInterface $list_source = NULL, array $filter_value = []): string {
+    $field_definition = $this->getFieldDefinition($facet, $list_source);
+    $field_type = !empty($field_definition) ? $field_definition->getType() : NULL;
+    if (in_array(EntityReferenceFieldItemListInterface::class, class_implements($field_definition->getClass()))) {
+      $entity_storage = $this->entityTypeManager->getStorage($field_definition->getSetting('target_type'));
+      $referenced_entities = $entity_storage->loadMultiple($filter_value);
+      return implode(', ', array_map(function ($referenced_entity) {
+        return $referenced_entity->label();
+      }, $referenced_entities));
+    }
+    elseif (in_array($field_type, ['list_integer', 'list_float', 'list_string'])) {
+      return implode(', ', array_map(function ($value) use ($field_definition) {
+        return $field_definition->getSetting('allowed_values')[$value];
+      }, $filter_value));
+    }
+    else {
+      return parent::getDefaultValuesLabel($facet, $list_source, $filter_value);
+    }
+  }
+
+  /**
+   * Gets field definition for the field used in the facet.
+   *
+   * @param \Drupal\facets\FacetInterface $facet
+   *   The facet.
+   * @param \Drupal\oe_list_pages\ListSourceInterface $list_source
+   *   The list source.
+   *
+   * @return \Drupal\Core\Field\FieldDefinitionInterface
+   *   The field definition.
+   */
+  protected function getFieldDefinition(FacetInterface $facet, ListSourceInterface $list_source): FieldDefinitionInterface {
+    $field_id = $list_source->getIndex()->getField($facet->getFieldIdentifier())->getOriginalFieldIdentifier();
+    $field_definitions = $this->entityFieldManager->getFieldDefinitions($list_source->getEntityType(), $list_source->getBundle());
+    return $field_definitions[$field_id];
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  public function buildDefaultValuesWidget(FacetInterface $facet, ListSourceInterface $list_source = NULL, array $parents = []): ?array {
+    $field_definition = $this->getFieldDefinition($facet, $list_source);
+    $field_type = !empty($field_definition) ? $field_definition->getType() : NULL;
+
+    if (in_array(EntityReferenceFieldItemListInterface::class, class_implements($field_definition->getClass()))) {
+      for ($i = 0; $i < count($facet->getActiveItems()) + 1; $i++) {
```

Comment:
Why are we counting the active items here? Why do we need to rely on the facets for building these form elements?

Reply:
The facet have the list of saved values and we count the list of active values for the edition of multiple values allowing also to add a new default value anytime the value is edited.